ui/src/telegrafs/components/CollectorCard.tsx

```diff
@@ -63,52 +53,46 @@ class CollectorRow extends PureComponent<Props & WithRouterProps> {
     const {collector, bucket} = this.props
 
     return (
-      <>
-        <IndexList.Row>
-          <IndexList.Cell>
-            <ComponentSpacer
-              margin={ComponentSize.Small}
-              direction={FlexDirection.Column}
-              alignItems={AlignItems.FlexStart}
-              stretchToFitWidth={true}
-            >
-              <EditableName
-                onUpdate={this.handleUpdateName}
-                name={collector.name}
-                noNameString={DEFAULT_COLLECTOR_NAME}
-                onEditName={this.handleNameClick}
-              />
-              <EditableDescription
-                description={collector.description}
-                placeholder={`Describe ${collector.name}`}
-                onUpdate={this.handleUpdateDescription}
-              />
-              {this.labels}
-            </ComponentSpacer>
-          </IndexList.Cell>
-          <IndexList.Cell>{bucket}</IndexList.Cell>
-          <IndexList.Cell revealOnHover={true} alignment={Alignment.Right}>
-            <ComponentSpacer
-              margin={ComponentSize.Small}
-              direction={FlexDirection.Row}
-              justifyContent={JustifyContent.FlexEnd}
-            >
-              <Button
-                text="Setup Instructions"
-                size={ComponentSize.ExtraSmall}
-                color={ComponentColor.Secondary}
-                onClick={this.handleOpenInstructions}
-              />
-              <ConfirmationButton
-                size={ComponentSize.ExtraSmall}
-                text="Delete"
-                confirmText="Confirm"
-                onConfirm={this.handleDeleteConfig}
-              />
-            </ComponentSpacer>
-          </IndexList.Cell>
-        </IndexList.Row>
-      </>
+      <ResourceList.Card
+        key={`telegraf-id--${collector.id}`}
+        testID="resource-card"
+        name={() => (
```

Comment:
I find these functions a little difficult to read, I wonder if we can return a function that returns JSX from a getter. then it could be name = {this.name} which is so much neater Would you like to give it a try? This is just aesthetic, so feel free to merge without it.